packages/core/src/shared/fs/fs.ts

```diff
@@ -211,6 +222,37 @@ export class FileSystem {
             return nodefs.rename(oldUri.fsPath, newUri.fsPath).catch(errHandler)
         }
 
+        /**
+         * We were seeing 'FileNotFound' errors during renames, even though we did a `writeFile()` right before the rename.
+         * The error looks to be from here: https://github.com/microsoft/vscode/blob/09d5f4efc5089ce2fc5c8f6aeb51d728d7f4e758/src/vs/platform/files/node/diskFileSystemProvider.ts#L747
+         * So a guess is that the exists()(stat() under the hood) call needs to be retried since there may be a race condition.
+         */
+        let attempts = 0
+        const isExists = await waitUntil(async () => {
+            const result = await fs.exists(oldUri)
+            attempts += 1
+            return result
+        }, renameTimeoutOpts)
+        // TODO: Move the `aws_fileSystem` or some variation of this metric in to common telemetry
+        if (!isExists) {
+            // source path never existed after multiple attempts.
+            // Either the file never existed, or had we waited longer it would have. We won't know.
+            telemetry.aws_fileSystem.emit({
```

Comment:
will this event be noisy? can we at least leave a comment like:

    // TODO: deduplicate

or add a field or something so that we can find these metrics later, when deduplication is generalized

Reply:
This should not be noisy as we only emit on the edge case where the source file does not exist or took time to appear. In the average case where it runs as normal, no telemetry is executed

Reply:
that might be true for now, but still should be a TODO since `aws_fileSystem` presumably could track other fs ops in the future

Reply:
Added a todo comment. We can probably have an underlying telemetry wrapper around all filesystem calls and then we report errors only. This will help uncover any unexpected failures.